src/Microsoft.AspNet.Mvc.Core/Description/DefaultApiDescriptionProvider.cs

```diff
@@ -450,5 +385,279 @@ private IApiResponseMetadataProvider[] GetResponseMetadataAttributes(ControllerA
+                //
+
+                var propertyCount = 0;
+                var unboundProperties = new HashSet<ModelMetadata>();
+
+                // We don't want to append the parameter name to the model name when recursing.
+                // Only append to the name if we're inside a container already.
+                if (modelMetadata.ContainerType != null)
+                {
+                    containerName = GetName(containerName, modelMetadata);
+                }
+
+                foreach (var propertyMetadata in modelMetadata.Properties)
+                {
+                    propertyCount++;
+                    var key = new PropertyKey(propertyMetadata, source);
+
+                    if (Visited.Add(key))
+                    {
+                        if (!Visit(propertyMetadata, source ?? ambientSource, containerName))
+                        {
+                            unboundProperties.Add(propertyMetadata);
+                        }
+                    }
+                    else
+                    {
+                        unboundProperties.Add(propertyMetadata);
+                    }
+                }
+
+                if (unboundProperties.Count == propertyCount && 
+                    source == null)
+                {
+                    // No properties were bound and we didn't find a new source, let the caller handle it.
+                    return false;
+                }
+                else
+                {
+                    foreach (var property in unboundProperties)
+                    {
+                        // Create a 'default' description for each property
+                        Context.Results.Add(CreateResult(property, source ?? ambientSource, containerName));
+                    }
+
+                    return true;
+                }
+            }
+
+            private ApiParameterDescription CreateResult(
+                ModelMetadata metadata,
+                ApiParameterSource source,
+                string containerName)
+            {
+                return new ApiParameterDescription()
+                {
+                    ModelMetadata = metadata,
+                    Name = GetName(containerName, metadata),
+                    ParameterDescriptor = Parameter,
+                    Source = source,
+                    Type = metadata.ModelType,
+                };
+            }
+
+            private static string GetName(string containerName, ModelMetadata metadata)
+            {
+                if (!String.IsNullOrEmpty(metadata.BinderModelName))
+                {
+                    // Name was explicitly provided
+                    return metadata.BinderModelName;
+                }
+                else
+                {
+                    return ModelBindingHelper.CreatePropertyModelName(containerName, metadata.PropertyName);
+                }
+            }
+
+            // This isn't extensible right now.
```

Comment:
`TryGetSource` is intentionally not extensible right now. We might want to add something like this in the future based on demand, but for now, our knowledge of what sources exist is hardcoded.